### PR TITLE
feat(console): modal layout

### PR DESCRIPTION
**Pull request by xiaoyijun**
## Summary

feat(console): modal layout

## Linear Issue Reference

* LOG-1907

## Testing

padding-top:
padding-top
padding-bottom:
padding-bottom
in the center:
center

### CODE REVIEW

File: packages/console/src/components/ModalLayout/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+import CardTitle from '../CardTitle';
+import IconButton from '../IconButton';
+import * as styles from './index.module.scss';
+
+type Props = {
+  title: AdminConsoleKey;
+  subtitle?: AdminConsoleKey;
+  hasFootDecoration?: boolean;
+  onClose: () => void;
+  content: () => ReactNode;
```
Comment (wangsijie) [inline packages/console/src/components/ModalLayout/index.tsx:17]: use children?

File: packages/console/src/components/ModalLayout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  footer: () => ReactNode;
```
Comment (wangsijie) [inline packages/console/src/components/ModalLayout/index.tsx:18]: I think a `ReactNode` is enough?
```
footer?: ReactNode;
```

File: packages/console/src/components/ModalLayout/index.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+import Card from '../Card';
+import CardTitle from '../CardTitle';
+import IconButton from '../IconButton';
+import * as styles from './index.module.scss';
+
+type Props = {
+  title: AdminConsoleKey;
+  subtitle?: AdminConsoleKey;
+  hasFootDecoration?: boolean;
```
Comment (wangsijie) [inline packages/console/src/components/ModalLayout/index.tsx:15]: set footer to optional is enough?

**Review by wangsijie — CHANGES_REQUESTED** [review]

**Comment by demonzoo** [comment]
This should've been applied to all modal dialogs. So perhaps add a follow-up task to handle that?

**Comment by xiaoyijun** [comment]
> This should've been applied to all modal dialogs. So perhaps add a follow-up task to handle that?

Added.
LOG-1934
@demonzoo

File: packages/console/src/components/ModalLayout/index.module.scss [hunk 4]
```diff
@@ -0,0 +1,17 @@
+@use '@/scss/underscore' as _;
+
+.container {
+  padding: _.unit(8);
+  max-width: 800px;
```
Comment (wangsijie) [inline packages/console/src/components/ModalLayout/index.module.scss:5]: Do we need a min width as well?

**Review by wangsijie — APPROVED** [review]
LGTM

File: packages/console/src/components/ModalLayout/index.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+import CardTitle from '../CardTitle';
+import IconButton from '../IconButton';
+import * as styles from './index.module.scss';
+
+type Props = {
+  title: AdminConsoleKey;
+  subtitle?: AdminConsoleKey;
+  children: ReactNode;
+  footer?: ReactNode;
+  onClose: () => void;
```
Comment (demonzoo) [inline packages/console/src/components/ModalLayout/index.tsx:16]: This should also be optional.

File: packages/console/src/components/ModalLayout/index.module.scss [hunk 6]
```diff
@@ -0,0 +1,17 @@
+@use '@/scss/underscore' as _;
+
+.container {
+  padding: _.unit(8);
```
Comment (demonzoo) [inline packages/console/src/components/ModalLayout/index.module.scss:4]: The container should have a max-height, say 80% or 90% of the screen height.
```css
max-height: 80vh;
```

File: packages/console/src/components/ModalLayout/index.module.scss [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+  .header {
+    display: flex;
+    align-items: flex-start;
+    justify-content: space-between;
+  }
+
+  .footer {
+    padding-top: _.unit(6);
+    border-top: 1px solid var(--color-neutral-90);
+  }
```
Comment (demonzoo) [inline packages/console/src/components/ModalLayout/index.module.scss:16]: header and footer should be unshrinkable, try `flex-shrink: 0`;

File: packages/console/src/components/ModalLayout/index.tsx [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+const ModalLayout = ({ title, subtitle, children, footer, onClose }: Props) => {
+  return (
+    <Card className={styles.container}>
+      <div className={styles.header}>
+        <CardTitle title={title} subtitle={subtitle} />
+        <IconButton onClick={onClose}>
+          <Close />
+        </IconButton>
+      </div>
+      <div>{children}</div>
```
Comment (demonzoo) [inline packages/console/src/components/ModalLayout/index.tsx:28]: `children` should be wrapped in a height flexiable content. Use:
```
flex: 1;
```

File: packages/console/src/components/ModalLayout/index.module.scss (hunk 4 above)
Comment (xiaoyijun) [reply packages/console/src/components/ModalLayout/index.module.scss:5]: LOG-1936

**Review by demonzoo — APPROVED** [review]

**Comment by demonzoo** [comment]
LGTM 👍

**Pull request merged**